components/notebook-controller/pkg/culler/culler.go

```diff
@@ -0,0 +1,194 @@
+package culler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("culler")
+var client = &http.Client{
+	Timeout: time.Second * 10,
+}
+
+// The constants with name 'DEFAULT_{ENV_Var}' are the default values to be
+// used, if the respective ENV vars are not present.
+// All the time numbers correspond to minutes.
+const DEFAULT_IDLE_TIME = "1440" // One day
+const DEFAULT_CULLING_CHECK_PERIOD = "1"
+const DEFAULT_ENABLE_CULLING = "false"
+const STOP_ANNOTATION = "kubeflow-resource-stopped"
+
+type NotebookStatus struct {
+	Started      string `json:"started"`
+	LastActivity string `json:"last_activity"`
+	Connections  int    `json:"connections"`
+	Kernels      int    `json:"kernels"`
+}
+
+// Some Utility Fuctions
+func getEnvDef(variable string, default_val string) string {
```

Comment:
def means default? seems a little weird

Reply:
Yes, on a second look it seems like I'm requesting the definition of the environment, which doesn't make a lot of sense.
Renamed it to `getEnvDefault` to make it a little more clear.